Content.Server/ParticleAccelerator/Components/ParticleAcceleratorControlBoxComponent.cs

```diff
@@ -325,11 +325,13 @@ public void RescanParts()
             _partEmitterCenter = null;
             _partEmitterRight = null;
 
+            var xform = _entMan.GetComponent<TransformComponent>(Owner);
+
             // Find fuel chamber first by scanning cardinals.
-            if (_entMan.GetComponent<TransformComponent>(Owner).Anchored)
+            if (xform.Anchored && _mapManager.TryGetGrid(xform.GridID, out var grid))
             {
-                var grid = _mapManager.GetGrid(_entMan.GetComponent<TransformComponent>(Owner).GridID);
```

Comment:
Okay so it was 5AM when I wrote this but how tf is it still anchored and on invalid grid.